pyccel/ast/cudatypes.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/devel/LICENSE for full license details.      #
+#------------------------------------------------------------------------------------------#
+""" Module containing types from the numpy module understood by pyccel
+"""
+from functools import lru_cache
+import numpy as np
+
+from .datatypes import FixedSizeNumericType, HomogeneousContainerType, PythonNativeBool
+from pyccel.utilities.metaclasses import ArgumentSingleton
+from .datatypes import pyccel_type_to_original_type, original_type_to_pyccel_type
+
+
+class CudaArrayType(HomogeneousContainerType, metaclass = ArgumentSingleton):
+    """
+    Class representing the Cuda array type.
+
+    Class representing the Cuda array type
+
+    dtype : NumpyNumericType | PythonNativeBool | GenericType
+        The internal datatype of the object (GenericType is allowed for external
+        libraries, e.g. MPI).
```

Comment:
"GenericType is allowed for external libraries, e.g. MPI" do you know if this is relevant for Cuda arrays?